wasmtime/_func.py

```diff
@@ -56,6 +63,36 @@ def type(self, store: Storelike) -> FuncType:
         ptr = ffi.wasmtime_func_type(store._context, byref(self._func))
         return FuncType._from_ptr(ptr, None)
 
+    def _create_raw_vals(self, *params: IntoVal) -> ctypes.Array[wasmtime_val_raw_t]:
+        raw = self._vals_raw_type()
+        for i, param_str in enumerate(self._params_str):
+            setattr(raw[i], param_str, params[i])
+        return raw
+    
+    def _extract_return(self, vals_raw: ctypes.Array[wasmtime_val_raw_t]) -> Union[IntoVal, Sequence[IntoVal], None]:
+        if self._results_n==0:
+            return None
+        if self._results_n==1:
+            return getattr(vals_raw[0], self._results_str0)
```

Comment:
I'm trying to move all the logic in the constructor (or in `_init_call`) so that this logic is run once the module is loaded not on every single call. and hence the speed boost.

I need [a minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) example about the two missing types, in a way similar to my SIMD example.

Reply:
I found the example it's `test_refs.py` I'll look into it and make sure it works